### PR TITLE
Caterer Page skeleton with some style

### PR DESCRIPTION
**Pull request by maxime-auneau**

### CODE REVIEW

**Review by axelhamil — DISMISSED** [review]

File: assets/styles/app.scss [hunk 1]
```diff
@@ -1,4 +1,6 @@
 @import 'variables';
+$primary: darken(#428bca, 20%);
+@import "~bootstrap/scss/bootstrap";
 
 * {
     box-sizing: border-box;
```
Comment (emrecaydin) [inline assets/styles/app.scss]: ok

File: config/packages/twig.yaml [hunk 2]
```diff
@@ -1,2 +1,3 @@
 twig:
     default_path: '%kernel.project_dir%/templates'
+    form_themes: ['bootstrap_4_layout.html.twig']
```
Comment (emrecaydin) [inline config/packages/twig.yaml:3]: ok

File: package.json [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
         "eslint-plugin-react": "^7.21.5",
-        "sass-lint": "^1.13.1"
+        "node-sass": "^5.0.0",
+        "regenerator-runtime": "^0.13.2",
+        "sass-lint": "^1.13.1",
+        "sass-loader": "^10.0.5",
+        "webpack-notifier": "^1.6.0"
     },
     "license": "UNLICENSED",
     "private": true,
```
Comment (emrecaydin) [inline package.json:18]: ok

File: package.json [hunk 4]
```diff
@@ -21,6 +22,5 @@
         "watch": "encore dev --watch",
         "build": "encore production --progress"
     },
-    "dependencies": {
-    }
+    "dependencies": {}
 }
```
Comment (emrecaydin) [inline package.json:26]: ok

**Review by axelhamil — DISMISSED** [review]

File: public/style/catererstyle.css [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+.text-quote-caterer{
+    text-align: center
+}
+
+p {
+    text-align: center;
+    align-self: center;
+}
```
Comment (emrecaydin) [inline public/style/catererstyle.css:37]: ok

File: src/Controller/CatererController.php [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+     * @Route("/caterer", name="caterer")
+     * @return Response
+     */
+    public function catererPage(): Response
+    {
+        $caterer = new Caterer();
+        $form = $this->createForm(CatererType::class, $caterer);
+        return $this->render('caterer/caterer.html.twig', ["form" => $form->createView()]);
+    }
+}
```
Comment (emrecaydin) [inline src/Controller/CatererController.php]: ok

File: src/Entity/Caterer.php [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+        return $this->description;
+    }
+
+    public function setDescription(string $description): self
+    {
+        $this->description = $description;
+
+        return $this;
+    }
+}
```
Comment (emrecaydin) [inline src/Entity/Caterer.php]: ok

File: src/Form/CatererType.php [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        ;
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'data_class' => Caterer::class,
+        ]);
+    }
+}
```
Comment (emrecaydin) [inline src/Form/CatererType.php]: ok

File: src/Repository/CatererRepository.php [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        return $this->createQueryBuilder('c')
+            ->andWhere('c.exampleField = :val')
+            ->setParameter('val', $value)
+            ->getQuery()
+            ->getOneOrNullResult()
+        ;
+    }
+    */
+}
```
Comment (emrecaydin) [inline src/Repository/CatererRepository.php:50]: ok

File: templates/base.html.twig [hunk 10]
```diff
@@ -5,7 +5,6 @@
     <link rel="icon" type="image/png" href="{{ asset('build/images/favicon.png') }}" />
     <title>{% block title %}{% endblock %}</title>
     {% block stylesheets %}
-        {{ encore_entry_link_tags('app') }}
     {% endblock %}
```
Comment (emrecaydin) [inline templates/base.html.twig]: ok

File: templates/base.html.twig [hunk 11]
```diff
@@ -5,7 +5,6 @@
...
 </head>
     <body>
```
Comment (emrecaydin) [inline templates/base.html.twig:9]: ok

File: templates/base.html.twig [hunk 12]
```diff
@@ -14,7 +13,6 @@
     {% endblock %}
 
     {% block javascripts %}
-        {{ encore_entry_script_tags('app') }}
     {% endblock %}
 
     </body>
```
Comment (emrecaydin) [inline templates/base.html.twig]: ok

File: templates/caterer/caterer.html.twig [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+{% form_theme form 'bootstrap_4_layout.html.twig' %}
+    {{ form_start(form) }}
+    {{ form_widget(form) }}
+    <input type='submit' value='Envoyer'>
+    {{ form_end(form) }}
+    {% block javascripts %}
+        {{ encore_entry_script_tags('app') }}
+    {% endblock %}
+{% endblock %}
```
Comment (emrecaydin) [inline templates/caterer/caterer.html.twig:50]: ok

File: templates/caterer/layout.html.twig [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+                <a href="#">Connexion</a>
+            </nav>
+            {% block headerTitle %}{% endblock %}
+        </div>
+        <hr>
+    </header>
+    <main>
+        {% block main %}{% endblock %}
+    </main>
+{% endblock %}
```
Comment (emrecaydin) [inline templates/caterer/layout.html.twig:22]: ok

File: yarn.lock [hunk 15]
```diff
@@ -1619,6 +1619,11 @@ boolbase@^1.0.0, boolbase@~1.0.0:
...
   integrity sha1-aN/1++YMUes3cl6p4+0xDcwed24=
 
+bootstrap@^4.5.3:
+  version "4.5.3"
+  resolved "https://registry.yarnpkg.com/bootstrap/-/bootstrap-4.5.3.tgz#c6a72b355aaf323920be800246a6e4ef30997fe6"
+  integrity sha512-o9ppKQioXGqhw8Z7mah6KdTYpNQY//tipnkxppWhPbiSWdD+1raYsnhwEZjkTHYbGee4cVQ0Rx65EhOY/HNLcQ==
+
 brace-expansion@^1.1.7:
   version "1.1.11"
   resolved "https://registry.yarnpkg.com/brace-expansion/-/brace-expansion-1.1.11.tgz#3c7fcbf529d87226f3d2f52b966ff5271eb441dd"
```
Comment (emrecaydin) [inline yarn.lock:1629]: ok

**Review by emrecaydin — DISMISSED** [review]
ok ok

**Review by axelhamil — DISMISSED** [review]

**Review by leobro7 — DISMISSED** [review]
ok

**Review by leobro7 — DISMISSED** [review]

**Review by axelhamil — DISMISSED** [review]

**Review by leobro7 — DISMISSED** [review]

File: src/Entity/Caterer.php [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+
+    /**
+     * @ORM\Column(type="string", length=150)
+     */
+    private string $budget;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    private string $date;
```
Comment (RomainClair) [inline src/Entity/Caterer.php]: This should probably be a date

File: src/Controller/CatererController.php [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+     * @Route("/caterer", name="caterer")
+     * @param Request $request
+     * @return Response
+     */
+    public function catererPage(Request $request): Response
+    {
+        $caterer = new Caterer();
+        $form = $this->createForm(CatererType::class, $caterer);
+        $form->handleRequest($request);
+        if ($form->isSubmitted()) {
```
Comment (RomainClair) [inline src/Controller/CatererController.php]: You should also validate you form, not only check for submission

**Review by RomainClair — CHANGES_REQUESTED** [review]
The Caterer entity lack the constraints for the validation process

**Review by axelhamil — DISMISSED** [review]

**Review by leobro7 — DISMISSED** [review]

File: src/Controller/CatererController.php [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+     * @param Request $request
+     * @return Response
+     */
+    public function catererPage(Request $request): Response
+    {
+        $caterer = new Caterer();
+        $form = $this->createForm(CatererType::class, $caterer);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $entityManager = $this->getDoctrine()->getManager();
```
Comment (RomainClair) [inline src/Controller/CatererController.php]: You should use the dependency injection instead

**Review by RomainClair — CHANGES_REQUESTED** [review]
The validator doesn't have any constraints to work with. You should add them in the Caterer entity

**Review by axelhamil — APPROVED** [review]

**Review by leobro7 — APPROVED** [review]

**Review by Mgaren — APPROVED** [review]

File: src/Entity/Caterer.php [hunk 19]
```diff
@@ -0,0 +1,195 @@
...
+    /**
+     * @ORM\Column(type="string", length=100)
+     * @Assert\NotBlank(message="Votre prénom ne peux pas être vide")
+     * @Assert\Length(max="100", maxMessage="Votre prénom ne peut pas dépasser 100 caractères")
+     */
+    private string $firstname;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     * @Assert\NotBlank(message="Votre ne Email ne peut pas être vide")
```
Comment (RomainClair) [inline src/Entity/Caterer.php:37]: May be there's too many "ne"

**Review by RomainClair — APPROVED** [review]

**Pull request merged**